python/tvm/relay/backend/contrib/tidl.py

```diff
@@ -0,0 +1,1383 @@
+        _tidlImportConv2d(conv2d_params, ctypes.POINTER(ctypes.c_int)())
+    
+        return True
+    
+    def tidl_import_pad(self, node):
+        r""" Import pad operator to TIDL
+            Get attributes pad_width, convert to array, and passs to C library.
+            A typical pad_width looks like: [[0,0],[0,1],[0,1],[0,0]]
+    
+        Parameters
+        ----------
+        node : relay.expr.Call
+            A relay.expr.Call node which is a pad operator
+    
+        Returns
+        -------
+        """
+    
+        pad_width = []
+        for i in range(len(node.attrs.pad_width)):
+            pad_width.append(get_const_tuple(node.attrs.pad_width[i]))
+        pad_list = [x for xs in pad_width for x in xs]
+    
+        # convert list to numpy array in order to pass to C library
+        pad_array = np.asarray(pad_list, dtype=np.int32)
+    
+        _tidlImportPad = self.import_lib.tidlImportPad
+        _tidlImportPad.argtypes = (ctypes.c_int, ctypes.c_void_p)
+        _tidlImportPad.restype  = None
+        _tidlImportPad(len(pad_array), ctypes.c_void_p(pad_array.ctypes.data))
+    
+    def tidl_import_add(self, node):
+        r""" Import add operator to TIDL
+            An "add" operator may be adding two nodes or adding one node with constant:
+                - %3 = add(%2, %1) 
+                - %3 = add(%2, %MobilenetV2/Conv/Conv2D_bn_offset) 
+            This function imports "add" opertor with args being two nodes.
+        """
+    
+        _tidlImportAdd = self.import_lib.tidlImportAdd
+        _tidlImportAdd.argtypes = None
+        _tidlImportAdd.restype  = None
+        _tidlImportAdd()
+    
+    def tidl_import_bias_add(self, node, params):
+        r""" Import bias_add or add operator to TIDL
+            An "add" operator may be adding two nodes or adding one node with constant:
+                - %3 = add(%2, %1) 
+                - %3 = add(%2, %MobilenetV2/Conv/Conv2D_bn_offset) 
+            A "bias_add" operator always add one node with constant.
+            This function imports a "bias_add" or "add" with args[1] being constant.
+    
+        Parameters
+        ----------
+        node : relay.expr.Call
+            A relay.expr.Call node which is a add operator
+        params : dict of str to tvm.NDArray
+            The parameter dict to be used by relay
+    
+        Returns
+        -------
+        True if import succeeds or False if import fails    
+        """
+        bias = node.args[1]
+        if isinstance(bias, tvm.relay.expr.Constant):
+            # bias is expr.Constant if bind_params_by_name is called
+            bias_params = bias.data
+        else:
+            # bias is expr.Var if bind_params_by_name is not called
+            print('bias_add op must have args[1] as expr.Constant')
+            return False
+    
+        if bias.checked_type.dtype == 'float32':
+            bias_params_dtype = b'float32'
+        #elif bias.checked_type.dtype == 'int8':

```

Comment:
Remove unused code, or I'm seeing this line a couple of times, is it planned to support this type? If so, probably put a TODO.

Reply:
Removed